### ISSUE TITLE
Vertical Pagination Iterator not working

### ISSUE THREAD

**Issue by Mr-EmPee** [RyseInventory/RyseInventory#68]
I've tried the new version and I've noticed a few problems.
 - The default value for the iterator type is vertical (I think that it should be horizontal)
 - It breaks after the first column is filled

Here is the code that I'm using to test it
```java
    Pagination pagination = contents.pagination();
    pagination.iterator(SlotIterator.builder()
        .type(SlotIterator.SlotIteratorType.VERTICAL)
        .startPosition(0, 1)
        .endPosition(4, 7)
        .blackList(8, 9, 17, 18, 26, 27)
        .build()
    );

    pagination.setItems(
        List.of(
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS)),
            IntelligentItem.empty(new ItemStack(Material.BLACK_STAINED_GLASS))
        )
    );
```

Result:
![Cattura](https://github.com/RyseInventory/RyseInventory/assets/86979852/6132156e-01f2-4fc8-adee-56a5a0a9f47a)

_Originally posted by @Mr-EmPee in https://github.com/RyseInventory/RyseInventory/issues/67#issuecomment-1575519411_

**Comment by Rysefoxx** [comment]
I take a look

**Comment by Rysefoxx** [comment]
Should be fixed in v1.6.4.
https://github.com/RyseInventory/RyseInventory/releases/tag/v1.6.4
